lib/types.js

```diff
@@ -2,11 +2,44 @@ const isNumber = val => typeof val === 'number';
 
 const castToNumber = val => {
   if(isNumber(val)) return val;
+
   const number = Number(val);
-  if(isNaN(number)) throw new CastError(Number, val);
+  if(isNaN(number)) { throw new CastError(Number, val);
+  }
+
   return number;
 };
 
+const isString = val => typeof val === 'string';
+
+const castToString = val => {
+  if(isString(val)) return val;
+  const string = String(val);
+  return string;
+};
+const isBoolean = val => typeof val === 'boolean';
+
+const castToBoolean = val => {
+  if(isBoolean(val)) return val;
+  const boolean = Boolean(val);
+  return boolean;
+};
+
+const isArray = val => Array.isArray(val);
+
+const castToArray = val => {
+  let array = Array(val);
+  if(Array.isArray(val)) {
+    return val;
+  } if(isNumber(val)) {
+    array = [val];
+  }
+
+  return array;
+};
+
+const isObject = val => typeof val === 'object';
```

Comment:
your isObject function could use more work. if i were to use this to check if an array is an object it would come back as true. for these labs we don't want that.